ddtrace/contrib/internal/ray/utils.py

```diff
@@ -121,26 +123,66 @@ def set_tag_or_truncate(span, tag_name, tag_value):
         span.set_tag(tag_name, tag_value)
 
 
-def get_dd_job_name_from_submission_id(submission_id: str):
+def get_dd_job_name_from_entrypoint(entrypoint: str):
     """
-    Get the job name from the submission id.
-    If the submission id is set but not in a job:test,run:3 format, return the default job name.
-    If the submission id is not set, return None.
+    Get the job name from the entrypoint.
     """
-    match = JOB_NAME_REGEX.match(submission_id)
+    match = ENTRY_POINT_REGEX.search(entrypoint)
     if match:
         return match.group(1)
     return None
 
 
-def get_dd_job_name_from_entrypoint(entrypoint: str):
+def json_to_dot_paths(data):
     """
-    Get the job name from the entrypoint.
+    Converts a JSON (or Python dictionary) structure into a dict mapping
+    dot-notation paths to leaf values, with keys prefixed once by RAY_METADATA_PREFIX.
+
+    - Assumes the top-level is a dictionary. If a list is encountered anywhere,
+      it is stringified with json.dumps and treated as a leaf (no recursion into list elements).
+    - Leaf values (str, int, float, bool, None) are returned as-is as the dict values.
+    - If the top-level value is a simple primitive (str, int, float, bool, None),
+      it is returned unchanged. If the top-level is a list, it is stringified.
+    - Returned dict keys are prefixed once with RAY_METADATA_PREFIX.
     """
-    match = ENTRY_POINT_REGEX.search(entrypoint)
-    if match:
-        return match.group(1)
-    return None
+
+    # If top-level is a list, stringify and return
+    if isinstance(data, list):
+        try:
+            return json.dumps(data, ensure_ascii=False)
+        except Exception:
+            return "[]"
+
+    flat = {}
+
+    def _recurse(node, path):
+        if isinstance(node, dict):
+            for key, value in node.items():
+                new_path = f"{path}.{key}" if path else key
+                _recurse(value, new_path)
+        elif isinstance(node, list):
+            # Treat any list as a leaf by stringifying it
+            try:
+                list_dump = json.dumps(node, ensure_ascii=False)
+            except Exception:
+                list_dump = "[]"
+            flat[path] = list_dump
+        else:
+            # leaf node: store the accumulated path -> value
+            flat[path] = node
+
+    _recurse(data, "")
+
+    # Prefix keys once with RAY_METADATA_PREFIX
+    result = {}
+    for k, v in flat.items():
+        if k:
+            result[f"{RAY_METADATA_PREFIX}.{k}"] = v
+        else:
+            # If an empty key (top-level primitive under a dict), attach prefix alone
+            result[RAY_METADATA_PREFIX] = v
+
+    return result
```

Comment:
if you use _recurse(data, RAY_METADATA_PREFIX) it would auto work I think without additional work.